common/dconfigclient/generate.go

```diff
@@ -74,6 +75,23 @@ func GenerateIntMaxMinVerifier(dconfigKey string, minV int, maxV int) Verifier {
 	return verifierFunc
 }
 
+// GenerateStringRegexpVerifier return a dconfig Verifier for the dconfigKey
+func GenerateStringRegexpVerifier(dconfigKey string, r *regexp.Regexp) Verifier {
+	verifierFunc := func(_ interface{}, newV interface{}) bool {
```

Comment:
I think you should make a local copy of `dconfigKey` and the regex expression, because the closure capture is by reference and you could end up with weird bugs.